sql/sql_select.cc

```diff
@@ -866,12 +866,6 @@ int SELECT_LEX::vers_setup_conds(THD *thd, TABLE_LIST *tables, COND **where_expr
       dst_cond= &table->on_expr;
     }
 
-    if (TABLE_LIST *t= table->embedding)
-    {
-      if (t->on_expr)
-        dst_cond= &t->on_expr;
-    }
```

Comment:
Justify removal of code. On what issue it was based?

Reply:
Code was added to fix [this issue](https://github.com/tempesta-tech/mariadb/issues/160). Now it seems unneeded. I don't know what exactly this code breaks because I don't understand how query optimization works.